pkg/cache/task.go

```diff
@@ -523,18 +516,16 @@ func (task *Task) releaseAllocation() {
 		s := TaskStates()
 		switch task.GetTaskState() {
 		case s.New, s.Pending, s.Scheduling, s.Rejected:
-			releaseRequest = common.CreateReleaseAskRequestForTask(
-				task.applicationID, task.taskID, task.application.partition)
+			releaseRequest = common.CreateReleaseRequestForTask(task.applicationID, task.taskID, task.allocationID, task.application.partition, task.terminationType)
```

Comment:
A single method to create a release request. From now on we'll always try to release both ask+allocation if possible.